src/tests/FastBoard_unittest.cpp

```diff
@@ -0,0 +1,342 @@
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(4, 3)));
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(4, 4))); 
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(5, 4)));
+}
+
+// Results will make more sense in FullBuard test
+TEST(FastBoardTest, IsSuicideWhenNotForBlack) {
+    FastBoard b;
+    b.reset_board(5);  
+    b.set_square(2, 2, FastBoard::WHITE);
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(1, 1), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(2, 1), FastBoard::BLACK));
+}
+
+// Results will make more sense in FullBuard test
+TEST(FastBoardTest, IsSuicideWhenForBlackInAllWhiteField) {
+    FastBoard b = create_5x5_all_white_field();
+
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(1, 1), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(3, 3), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(4, 4), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(4, 2), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(4, 4), FastBoard::BLACK));
+}
+
+TEST(FastBoardTest, CalcAreaScore) {
+    FastBoard b = create_filled_5x5();
+    EXPECT_EQ(-6.5, b.area_score(6.5F)); 
+    EXPECT_EQ(-.5, b.area_score(0.5F));
+    EXPECT_EQ(-9.0, b.area_score(9.0F)); 
+}
+
+TEST(FastBoardTest, CalcAreaScoreOnWhiteField) {
+    FastBoard b = create_5x5_all_white_field();
+    EXPECT_EQ(-31.5, b.area_score(6.5F)); 
+    EXPECT_EQ(-25.5, b.area_score(0.5F));
+    EXPECT_EQ(-34.0, b.area_score(9.0F)); 
+}
+
+TEST(FastBoardTest, CalcAreaScoreOnSemiFilled9x9) {
+    FastBoard b = create_filled_9x9();
+    EXPECT_EQ(-7.5, b.area_score(6.5F)); 
+    EXPECT_EQ(-1.5, b.area_score(0.5F));
+    EXPECT_EQ(-10.0, b.area_score(9.0F)); 
+}
+
+TEST(FastBoardTest, ToMove) {
+    FastBoard b = create_filled_5x5();
+    EXPECT_EQ(FastBoard::BLACK, b.get_to_move());
+    EXPECT_EQ(true, b.black_to_move());
+    b.set_to_move(FastBoard::WHITE);
+    EXPECT_EQ(FastBoard::WHITE, b.get_to_move());
+    EXPECT_EQ(false, b.black_to_move());
+}
+
+TEST(FastBoardTest, MoveToText) {
+    FastBoard b = create_filled_3x3();
+    EXPECT_EQ("B1", b.move_to_text(b.get_vertex(1, 0)));
+    EXPECT_EQ("A2", b.move_to_text(b.get_vertex(0, 1)));
+    EXPECT_EQ("pass", b.move_to_text(FastBoard::PASS));
+    EXPECT_EQ("resign", b.move_to_text(FastBoard::RESIGN));
+}
+
+TEST(FastBoardTest, MoveToTextSgf) {
+    FastBoard b = create_filled_3x3();
+    EXPECT_EQ("bc", b.move_to_text_sgf(b.get_vertex(1, 0)));
+    EXPECT_EQ("ab", b.move_to_text_sgf(b.get_vertex(0, 1)));
+    EXPECT_EQ("ca", b.move_to_text_sgf(b.get_vertex(2, 2)));
+    EXPECT_EQ("tt", b.move_to_text_sgf(FastBoard::PASS));
+    EXPECT_EQ("tt", b.move_to_text_sgf(FastBoard::RESIGN));
+}
+
+TEST(FastBoardTest, GetStoneList) {
+    FastBoard emptyBoard;
+    emptyBoard.reset_board(3);
+    EXPECT_EQ("", emptyBoard.get_stone_list());
+    
+    FastBoard b = create_filled_5x5();
+    EXPECT_EQ("A4 B2 C2 C3 C4 C5 D2 D3", b.get_stone_list());
```

Comment:
What if the order of the stones changes? Why would that be wrong?

Reply:
It is true that if the implementation changed the order of stones in get_stone_list, then the test would fail, and it probably should not. But if the order of the stones does not matter, maybe that method should return a Set instead of a string. Then checking varying order would be easier. As it is now it is more of a regression test rather than a unit test. It would have to be updated if the order changed. Possible resolutions:
   1) leave it as is and accept that it is only a regression test.
   2) remove it and not test this method.
   3) add a get_stone_set that returns a set of stones and is unite tested, and a separate get_stone_list method that calls get_stone_set and converts the set to a string. Unit test the get_stone_set method.
   4) Convert the space delimited string to a set and then compare it with a set expectation. While this would be most correct, it would also mean adding quite a bit of test logic.
   5) Some other suggestion
   My preference is for 1).